crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -131,26 +140,59 @@ impl QuotedOrder {
 
     /// The UID of the order.
     pub fn order_uid(&self, blockchain: &Blockchain) -> tests::boundary::OrderUid {
-        self.boundary(blockchain).uid()
+        self.boundary(
+            blockchain,
+            blockchain.trader_secret_key,
+            blockchain.trader_address,
+        )
+        .uid()
     }
 
     /// The signature of the order.
     pub fn order_signature(&self, blockchain: &Blockchain) -> Vec<u8> {
-        self.boundary(blockchain).signature()
+        self.boundary(
+            blockchain,
+            blockchain.trader_secret_key,
+            blockchain.trader_address,
+        )
+        .signature()
+    }
+
+    /// The signature of the order by surplus capturing JIT-order owner
+    pub fn order_signature_by_surplus_capturing_jit_order_owner(
```

Comment:
Seems strange that `QuotedOrder`s would have any relation to JIT orders. Seems more appropriate to have a `sign()` function on the inner `Order` type instead.

Reply:
It has the exact same relation as normal orders, except for the fields `buy` and `sell`. It follows the exact same principle. If I move that function into `Order` then I would need to move also the function for signing normal orders too.

Reply:
My point is that conceptually it seems odd that a JIT order would have to be quoted in some way. JIT orders can make up any limit prices they want whereas the `QuotedOrder` struct refers to a trade a user wants to do which currently has to be quoted before it can happen.

Specifically the doc comment on `QuotedOrder` says:
> /// An order for which buy and sell amounts have been calculated.

> If I move that function into Order then I would need to move also the function for signing normal orders too.

Yeah. When we add a feature to a piece of code that was written without that unforeseen feature in mind we have to refactor the old code to make the new feature fit nicely. Otherwise we'll drown in flags, workarounds and hotfixes eventually.

Overall there are significant conceptual differences between user orders and JIT orders despite sharing the same fields. For the same reason there are new types like `Address` and `ContractAddress` where they are both just `H160` under the hood but their meaning is different enough that they warrant a new type.